src/Parlot/Character.cs

```diff
@@ -45,7 +45,7 @@ public static bool IsWhiteSpaceOrNewLine(char ch)
         public static bool IsNewLine(char ch)
             => (ch == '\n') || (ch == '\r') || (ch == '\v');
 
-        public static char ScanHexEscape(string text, int index, out int length)
+        public static char ScanHexEscape(char[] text, int index, out int length)
```

Comment:
why not `ScanHexEscape(BufferSpan<char> text, int index, out int length)` so it doesn't have to allocate a `char[]`

Reply:
it will have to allocate a char[] as soon as you will have an escape

Reply:
but here it has to allocate two of them, the argument, and the returned value.